### PR TITLE
[Backport stable/8.1] Ban process instances through admin API

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

Manual backport of #13550

### CODE REVIEW

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionAdminAccess.java [hunk 1]
```diff
@@ -140,4 +149,91 @@ public ActorFuture<Void> resumeProcessing() {
...
+                            "Could not retrieve writer to write error record for process instance.",
+                            error);
+                        future.completeExceptionally(error);
+                        return;
+                      }
+
+                      writeErrorEventAndBanInstance(processInstanceKey, writer, future);
+                    });
+          } catch (final Exception e) {
+            LOG.error("Could not resume processing", e);
```
Comment (oleschoenburg) [inline broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionAdminAccess.java:175]: :wrench: I missed this for the PR on `main` but this error message is not correct. Could you adjust it here and on main?

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionAdminAccess.java [hunk 2]
```diff
@@ -140,4 +149,91 @@ public ActorFuture<Void> resumeProcessing() {
...
+        RecordBatchEntry.createEntry(
+            processInstanceKey,
+            -1,
+            recordMetadata.getRecordType(),
+            recordMetadata.getIntent(),
+            recordMetadata.getRejectionType(),
+            recordMetadata.getRejectionReason(),
+            recordMetadata.getValueType(),
+            errorRecord);
+    writer.key(processInstanceKey);
```
Comment (oleschoenburg) [inline broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionAdminAccess.java:236]: :wrench: Can be part of the method call chain one line below

**Review by lenaschoenburg — APPROVED** [review]
Nice work @rodrigo-lourenco-lopes :clap:  
Please consider my review comments before merging.

**Comment by rodrigo-lourenco-lopes** [comment]
bors merge

**Pull request merged**